extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.js

```diff
@@ -173,19 +176,49 @@ export class AmpViewerAssistance {
       return false;
     }
     const update = args['update'];
+    const error = args['error'];
     const actionStatus = update && update['actionStatus'];
-    if (!update || !actionStatus) {
+
+    if (error) {
+      return true;
+    } else if (update) {
+      if (!actionStatus || !ACTION_STATUS_WHITELIST.includes(actionStatus)) {
+        user().error(TAG, '"updateActionState" action "update" parameter' +
+            ' must contain a valid "actionStatus" field.');
+        return false;
+      }
+      return true;
+    } else {
       user().error(TAG, '"updateActionState" action must have an' +
-          ' an "update" parameter containing an "actionStatus" field.');
+          ' "update" or "error" parameter.');
       return false;
     }
-    if (!ACTION_STATUS_WHITELIST.includes(actionStatus)) {
-      user().error(TAG, 'Invalid "update.actionStatus" value for ' +
-          '"updateActionState":', actionStatus);
-      return false;
+  }
+
+  /**
+   * Transforms an error response object into a suitable updateActionState
+   * payload to send to the viewer.
+   * @private
+   * @param {?JsonObject} args
+   * @return {!Promise<!JsonObject>}
+   */
+  transformUpdateActionStateArgs_(args) {
+    if (args['update']) {
+      return Promise.resolve(args);
     }
-    user().info(TAG, 'Sending "actionStatus":', actionStatus);
-    return true;
+    // Must transform 'error' Response object
+    const response = args['error'];
+    return response.text().then(errorMessage => {
+      return {
```

Comment:
Let's wrap this outermost object in `dict()` call to ensure `JsonObject` type. Also, use strings for object keys.

Context: We're working on more aggressive code minification via property renaming. JsonObject and quoted string keys opt-out of renaming.

Reply:
Done.